File: Data Structures & Algorithms/duplicate-integer/submission-0.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        Dictionary<int, bool> existingDictionary = new Dictionary<int, bool>();

        for (int i = 0; i < nums.Length; i++)
        {
            existingDictionary.TryGetValue(nums[i], out var exists);

            if (exists)
            {
                return true;
            }
            else
            {
                existingDictionary.Add(nums[i], true);
            }
        }

        return false;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        output = []

        for i in range(0, len(nums)):
            el = 1
            for j in range(0, len(nums)):
                if i == j:
                    continue

                el *= nums[j]
            output.append(el)

        return output

                